sdk/core/azure-core/CMakeLists.txt

```diff
@@ -29,11 +27,25 @@ add_library (
   src/http/transport_policy.cpp
   src/http/telemetry_policy.cpp
   src/http/url.cpp
-  src/http/winhttp/win_http_transport.cpp
   src/logging/logging.cpp
   src/strings.cpp
   src/version.cpp)
 
+if(BUILD_CURL_TRANSPORT)
+  set(
+    AZURE_CORE_CURL_TRANSPORT_SOURCE
+    src/http/curl/curl.cpp
+  )
+endif()
+if(BUILD_WINHTTP_TRANSPORT)
+  set(
+    AZURE_CORE_WINHTTP_TRANSPORT_SOURCE
+    src/http/winhttp/win_http_transport.cpp
+  )
+endif()
+
+add_library (${TARGET_NAME} ${AZURE_CORE_SOURCE} ${AZURE_CORE_CURL_TRANSPORT_SOURCE} ${AZURE_CORE_WINHTTP_TRANSPORT_SOURCE})
```

Comment:
nit/questions: What is the string value of `${AZURE_CORE_WINHTTP_TRANSPORT_SOURCE}` when `BUILD_WINHTTP_TRANSPORT` is false?

 I am guessing it's empty. Is having an empty `''` parameter in `add_library` OK to do?

Maybe there is a way to chain the calls to `add_library` and only call `add_library ${AZURE_CORE_WINHTTP_TRANSPORT_SOURCE}` inside the above if-block.